plugins/nf-amazon/src/main/com/upplication/s3fs/S3OutputStream.java

```diff
@@ -641,14 +649,11 @@ int getPartsCount() {
      */
     static synchronized ExecutorService getOrCreateExecutor(int maxThreads) {
         if( executorSingleton == null ) {
-            ThreadPoolExecutor pool = new ThreadPoolExecutor(
-                    maxThreads,
-                    Integer.MAX_VALUE,
-                    60L, TimeUnit.SECONDS,
-                    new LimitedQueue<Runnable>(maxThreads *3),
-                    new ThreadPoolExecutor.CallerRunsPolicy() );
-
-            pool.allowCoreThreadTimeOut(true);
+            ThreadPoolExecutor pool = ThreadPoolBuilder.io(
+                    1,
+                    maxThreads*3,
```

Comment:
still some concerns on `maxThreas * 3`. currently the default value of `maxThreads` is the following 

https://github.com/nextflow-io/nextflow/blob/ac63503fe2a2fae19d7f21530b69b2198669e087/plugins/nf-amazon/src/main/com/upplication/s3fs/util/S3MultipartOptions.java#L63-L63


therefore actual value specified is `cpus * 3 * 3` is this really wanted? even so I think this should in the `maxThreads` initialization, so that when providing an explicit value via the config file it does not get altered.

Reply:
yes, you're right, I will fix it

related: `maxThreads*10`  was an arbitrary number I took after some test but I'm thinking maybe we need to maintain current logic ( `new LimitedQueue<Runnable>(maxThreads *3)`)

Reply:
In principle when the tasks in the pool are > maxThreads they just are queued an wait. That `LimitedQueue` is a kind of hack. But please test it

Reply:
I've tested uploading a 100Gb file in a container with 3Gb of memory and configuration:

```  
    uploadChunkSize = 20.Mb
    uploadMaxThreads = 2
```

- with a pool of maxThreads*3 it spent 1h 10mnts
- with a pool of maxThreads*10 it spent 48 mnts

so, I think a pool of x10 can be a valid value as it allows to reduce the upload time and use the same amount of memory

Reply:
I've run a couple of tests in a container with 3Gb of memory and it took ~1h to upload a 100Gb file using a `maxThreads*3 `  and `maxThreads*10` queue size in both cases so probably with a x3 pool is enough